elm/wizard.py

```diff
@@ -87,6 +90,9 @@ def engineer_query(self, query, token_budget=None, new_info_threshold=0.7,
         references : list
             The list of references (strs) used in the engineered prompt is
             returned here
+        vector_query_time : float
```

Comment:
I know you didnt do this but can you add a docstring for `used_index` here? not sure why we dont have one.